### PR TITLE
Addition of Concode Dataset 

### PR DESCRIPTION
**Pull request by reshinthadithyan**
##Overview
Concode Dataset contains pairs of Nl Queries and the corresponding Code.(Contextual Code Generation)

Reference Links
Paper Link = https://arxiv.org/pdf/1904.09086.pdf
Github Link =https://github.com/microsoft/CodeXGLUE/tree/main/Text-Code/text-to-code

### CODE REVIEW

**Comment by reshinthadithyan** [comment]
Testing command doesn't work
###trace
-- Docs: https://docs.pytest.org/en/stable/warnings.html
========================================================= short test summary info ========================================================== 
ERROR tests/test_dataset_common.py - absl.testing.parameterized.NoTestsError: parameterized test decorators did not generate any tests. Ma...
====================================================== 2 warnings, 1 error in 54.23s ======================================================= 
ERROR: not found: G:\Work Related\hf\datasets\tests\test_dataset_common.py::LocalDatasetTest::test_load_real_dataset_concode
(no name 'G:\\Work Related\\hf\\datasets\\tests\\test_dataset_common.py::LocalDatasetTest::test_load_real_dataset_concode' in any of [<Module test_dataset_common.py>])

**Comment by reshinthadithyan** [comment]
Hello @lhoestq Test checks are passing in my local, but the commit fails in ci. Any idea onto why?   
#### Dummy Dataset Test 
====================================================== 1 passed, 6 warnings in 7.14s ======================================================= 
#### Real Dataset Test 
====================================================== 1 passed, 6 warnings in 25.54s ======================================================

File: datasets/codexglue_concode/codexglue_concode.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+  year=2018,
+}
+"""
+_URL = "https://raw.githubusercontent.com/microsoft/CodeXGLUE/main/Text-Code/text-to-code/dataset/concode/"
+_URLS = {
+    "train": _URL + "train.json",
+    "dev": _URL + "dev.json",
+    "test": _URL + "test.json",
+}
+_DESCRIPTION = """Mapping Language to Code in a Programmatic Context"""
```
Comment (lhoestq) [inline datasets/codexglue_concode/codexglue_concode.py:20]: Could you add more details ? Where does the dataset come from ? How big is it etc.

File: datasets/codexglue_concode/codexglue_concode.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+            version=VERSION,
+            description="Plain text",
+        ),
+    ]
+    def _info(self):
+            return datasets.DatasetInfo(
+                description=_DESCRIPTION,
+                features=datasets.Features(
+                    {
+                        "nl": datasets.Value("string"),
```
Comment (lhoestq) [inline datasets/codexglue_concode/codexglue_concode.py:48]: what's nl exactly ?
If it is the language name, maybe we should name it `language` and use ClassLabel feature type ?

File: datasets/codexglue_concode/codexglue_concode.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+        Args:
+          filename: name of the file to be read.
+
+
+        Yields:
+          A dictionary of features.
+        """
+        logging.info("Generating examples from from %s split ",filename)
+        id = 0
+        with open(filename) as f:
```
Comment (lhoestq) [inline datasets/codexglue_concode/codexglue_concode.py:88]: please specify the encoding here. Ex: `encoding="utf-8"`

**Review by lhoestq — COMMENTED** [review]
That's really cool thanks !

First to fix the CI you have to format the code using `make style`, and add the encoding in the `open` call

**Comment by reshinthadithyan** [comment]
Hello @lhoestq, Have a look, I've changed according to the reviews. Thanks!

**Comment by yjernite** [comment]
@reshinthadithyan that's a great start! You will also need to add a Dataset card, following the instructions given here

**Comment by reshinthadithyan** [comment]
> @reshinthadithyan that's a great start! You will also need to add a Dataset card, following the instructions given [here](https://github.com/huggingface/datasets/blob/master/ADD_NEW_DATASET.md#manually-tag-the-dataset-and-write-the-dataset-card)

Hello @yjernite I'm facing issues in using the datasets-tagger Refer #1 in datasets-tagger. Thanks

**Comment by yjernite** [comment]
> > @reshinthadithyan that's a great start! You will also need to add a Dataset card, following the instructions given [here](https://github.com/huggingface/datasets/blob/master/ADD_NEW_DATASET.md#manually-tag-the-dataset-and-write-the-dataset-card)
> 
> Hello @yjernite I'm facing issues in using the datasets-tagger Refer #1 in datasets-tagger. Thanks

Hi @reshinthadithyan ! Did you try with the latest version of the tagger? What issues are you facing?

We're also relaxed the dataset requirement for now, you'll only add to add the tags :)

**Comment by lhoestq** [comment]
Could you work on another branch when adding different datasets ?
The idea is to have one PR per dataset

**Comment by lhoestq** [comment]
Thanks ! The github diff looks all clean now :) 
To fix the CI you just need to rebase from master

Don't forget to add the tags of the dataset card. It's the yaml part at the top of the dataset card
More infor here : https://github.com/huggingface/datasets/blob/master/ADD_NEW_DATASET.md#tag-the-dataset-and-write-the-dataset-card

The issue you had with the tagger should be fixed now by https://github.com/huggingface/datasets-tagging/pull/5

**Pull request closed without merging**